Packs/FetchIndicatorsFromFile/Scripts/FetchIndicatorsFromFile/FetchIndicatorsFromFile.py

```diff
@@ -47,21 +47,39 @@ def xls_file_to_parsed_indicators(file_path, sheet_name, indicator_col, type_col
     return indicator_list
 
 
-def txt_file_to_indicator_list(file_path):
+def txt_file_to_indicator_list(file_path, auto_detect, default_type):
     with open(file_path, "r") as fp:
         file_data = fp.read()
 
     indicator_list = re.split('\n|,|, ', file_data)
-    return indicator_list
+
+    return parse_indicators(indicator_list, auto_detect, default_type)
+
+
+def parse_indicators(indicator_list, auto_detect, default_type):
+    parsed_indicators = []
+    for indicator in indicator_list:
+        if auto_detect:
+            indicator_type = detect_type(indicator)
+
+        else:
+            indicator_type = default_type
+
+        parsed_indicators.append({
+            "Type": indicator_type,
+            "Value": indicator
+        })
+
+    return parsed_indicators
```

Comment:
you can do this on line 22 or 44, not sure why we need this logic..

Reply:
done